sdk/ios/Configsum/Configsum/Configsum/Configsum.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  Configsum.swift
+//  Configsum
+//
+//  Created by Alexandru Savu on 2017-10-04.
+//  Copyright © 2017 Lifesum. All rights reserved.
+//
+
+public typealias failureBlock = (Error) -> Void
+public typealias successBlock = (Bool) -> Void
+
+public struct Configsum {
+    private let dispatcher: Dispatcher
+
+    public init(environment: Environment,
+                token: String,
+                authorizationHeader: String) {
+        let httpClient = HTTPClient(environment: environment)
+        let persistence = Persistence(environment: environment)
+        self.dispatcher = Dispatcher(httpClient: httpClient,
+                                     persistence: persistence)
+    }
+    
+    public func getBool(key: String,
+                        defaultValue: Bool) throws -> Bool {
+        return try dispatcher.getValue(forKey: key,
+                            defaultValue: defaultValue)
+    }
+    public func getString(key: String,
```

Comment:
this function will return a value based on the provided key or a default value in case the provided key is invalid. it throws only when the typecast fails. I think this error should be propagated to the consumer as they might get a different type then they expect which might break their assumptions. let me know if this sounds good to you

Reply:
is it possible to return something other than the specified return type? in this case `String`

 you can always log the stuffs, but it is a  design decision, how should the client react to such a situation? for example do you want to crash the app or do you want to fall back to default value?